core/src/main/java/org/elasticsearch/transport/TransportStatus.java

```diff
@@ -56,4 +57,15 @@ public static byte setCompress(byte value) {
         value |= STATUS_COMPRESS;
         return value;
     }
+
+    static boolean isHandshake(byte value) { // pkg private since it's only used internally
+        return (value & STATUS_HANDSHAKE) != 0;
+    }
+
+    static byte setHandshake(byte value) { // pkg private since it's only used internally
```

Comment:
hmm I had to click "load diff" on that TcpTransport for some reason... it's here https://github.com/elastic/elasticsearch/pull/22094/files#diff-2ca1a68dd30dbf90e25a495b98f2cbccR1473